### PR TITLE
DI-203 Connect processor event lambda to the non-prod dos db read replica

### PR DESCRIPTION
**Pull request by mabe13**
## Description

This PR will change the DoS database the event processor lambda to connect to the DoS regression read replica in non-prod for DoS Integration. To do this it will also add a new make target to check the connection is working with an example change event payload that can be passed to the lambda.  

### Noteworthy Changes

- These are changes the reviewer should look out for
- New make target `make test-deployed-event-processor-db-connection` for testing with example change event for ods code that exists in the replica database (**see ReadMe to see how to run it**)
- Changes to the event processor to select the right schema
- Changes to get username from secret manager and change to get database RDS endpoint from AWS CLI through new make target

## Type of change

- New feature (non-breaking change which adds functionality)

## Testing

Please tick the testing that has been completed

- [ ] Unit tests
- [ ] Component tests

## Developer Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have run the [code formatting checks](../README.md#code-quality)
- [ ] I have run the [code quality checks](../README.md#code-quality)
- [ ] New code meets [standards](https://nhsd-confluence.digital.nhs.uk/display/DI/DI+Ways+of+Working) agreed by the team
- [ ] Unit test code coverage is at or above 80%
- [ ] New and existing unit tests pass locally with my changes
- [ ] Tests have added that prove my fix is effective or that my feature works (Component tests)
- [ ] I have made corresponding changes to the documentation
- [ ] I have cleaned down my environment (if created)

## Code Reviewer Checklist

- [ ] I have run the unit tests and they run correctly
- [ ] I can confirm the changes have been tested or approved by a tester
- [ ] I can confirm no remaining infrastructure is left over from this branch

### CODE REVIEW

**Review by SLClayton — DISMISSED** [review]

**Review by rohith1122 — DISMISSED** [review]
Didn't  test  but it LGTM

**Review by dehe1 — DISMISSED** [review]

**Review by ed-sparkes — CHANGES_REQUESTED** [review]

**Review by ed-sparkes — DISMISSED** [review]

File: build/automation/lib/aws.mk [hunk 1]
```diff
@@ -236,7 +236,14 @@ aws-rds-describe-instance: ### Describe RDS instance - mandatory: DB_INSTANCE
 		$(AWSCLI) rds describe-db-instances \
 			--region $(AWS_REGION) \
 			--db-instance-identifier=$(DB_INSTANCE) \
-	" | make -s docker-run-tools CMD="jq -r '.DBInstances[0]'"
+	" | make -s docker-run-tools CMD="jq -r '.DBInstances[0].Endpoint.Address'"
```
Comment (JackPlowman) [inline build/automation/lib/aws.mk:239]: Is this line expected to be changed?

**Review by JackPlowman — CHANGES_REQUESTED** [review]
LGMT, not sure about one change which I left a comment for. Once that it resolved I'm happy to approve

File: build/automation/lib/aws.mk (hunk 1 above)
Comment (mabe13) [reply build/automation/lib/aws.mk]: good spot missed it, made the change to test the function added before creating it 👍

**Review by JackPlowman — APPROVED** [review]
LGTM 👍

**Review by ed-sparkes — APPROVED** [review]

**Pull request merged**